src/home/celebration/NftCelebration.tsx

```diff
@@ -0,0 +1,233 @@
+    (props: BottomSheetDefaultBackdropProps) => (
+      <BottomSheetBackdrop {...props} disappearsOnIndex={-1} appearsOnIndex={0} />
+    ),
+    []
+  )
+
+  const renderHandleWithImage = useCallback(
+    () => (
+      <View style={styles.handleWithImage}>
+        {matchingNft && (
+          <NftMedia
+            shouldAutoScaleHeight
+            nft={matchingNft}
+            ErrorComponent={
+              <View style={styles.imageError}>
+                <ImageErrorIcon />
+              </View>
+            }
+            origin={NftOrigin.NftCelebration}
+            mediaType="image"
+          />
+        )}
+        <View style={styles.handleBar} />
+      </View>
+    ),
+    [matchingNft]
+  )
+
+  useEffect(() => {
+    if (isVisible) {
+      // Wait for the home screen to have less ongoing async tasks.
+      // This should help the bottom sheet animation run smoothly.
+      const timeoutId = setTimeout(() => {
+        bottomSheetRef.current?.expand()
+      }, 1000)
+      return () => clearTimeout(timeoutId)
+    }
+  }, [isVisible])
+
+  const handleBottomSheetPositionChange = (index: number) => {
+    if (!matchingNft) {
+      return // this should never happen
+    }
+
+    if (index === -1) {
+      ValoraAnalytics.track(HomeEvents.nft_celebration_displayed, {
+        networkId: matchingNft.networkId,
+        contractAddress: matchingNft.contractAddress,
+      })
+
+      vibrateSuccess()
+
+      confettiStartTime.current = Date.now()
+      setShowConfetti(true)
+    }
+  }
+
+  const handleCtaPress = () => {
+    bottomSheetRef.current?.close()
+  }
+
+  const handleConfettiFinish = () => {
+    finishCelebration({ userInterrupted: false })
+  }
+
+  const handleConfettiDismiss = () => {
+    finishCelebration({ userInterrupted: true })
+  }
+
+  const finishCelebration = ({ userInterrupted }: { userInterrupted: boolean }) => {
+    if (!matchingNft) {
+      return // this should never happen
+    }
+
+    ValoraAnalytics.track(HomeEvents.nft_celebration_animation_displayed, {
+      userInterrupted,
+      durationInSeconds: Math.round((Date.now() - confettiStartTime.current) / 1000),
+    })
+
+    setShowConfetti(false)
+
+    dispatch(nftCelebrationDisplayed())
+  }
+
+  if (!isVisible) {
+    return null
+  }
+
+  return (
+    <>
+      <GorhomBottomSheet
+        ref={bottomSheetRef}
+        enableDynamicSizing
+        enablePanDownToClose
+        index={-1}
+        backdropComponent={renderBackdrop}
+        handleComponent={renderHandleWithImage}
+        backgroundStyle={styles.bottomSheetBackground}
+        onChange={handleBottomSheetPositionChange}
+      >
+        <BottomSheetView style={[styles.container, insetsStyle]}>
+          <View style={styles.content}>
+            <Text style={styles.title}>{t('nftCelebration.bottomSheet.title')}</Text>
+            <Text style={styles.description}>
+              {t('nftCelebration.bottomSheet.description', { appName: APP_NAME })}
```

Comment:
nit: appName comes for free [here](https://github.com/valora-inc/wallet/blob/alex/gift-celebration/src/i18n/index.ts#L57) so you don't need to pass it in here :)